src/main/java/com/minecolonies/core/entity/ai/workers/production/EntityAIStructureMiner.java

```diff
@@ -951,17 +970,33 @@ public boolean shallReplaceSolidSubstitutionBlock(final Block worldBlock, final
     }
 
     @Override
-    protected void triggerMinedBlock(@NotNull final BlockState blockToMine)
+    protected void triggerMinedBlock(@NotNull final BlockPos position, @NotNull final BlockState blockToMine)
     {
-        super.triggerMinedBlock(blockToMine);
-
-        final double chance = 1 + worker.getCitizenColonyHandler().getColony().getResearchManager().getResearchEffects().getEffectStrength(MORE_ORES);
+        super.triggerMinedBlock(position, blockToMine);
 
         if (IColonyManager.getInstance().getCompatibilityManager().isLuckyBlock(blockToMine.getBlock()))
         {
-            final int level = building.getBuildingLevel();
-            InventoryUtils.transferItemStackIntoNextBestSlotInItemHandler(IColonyManager.getInstance().getCompatibilityManager().getRandomLuckyOre(chance, level),
-              worker.getInventoryCitizen());
+            final double chance = 1 + worker.getCitizenColonyHandler().getColony().getResearchManager().getResearchEffects().getEffectStrength(MORE_ORES);
+            final boolean canGetLuckyBlock =
+              worker.getRandom().nextDouble() * ONE_HUNDRED_PERCENT <= MinecoloniesAPIProxy.getInstance().getConfig().getServer().luckyBlockChance.get() * chance;
+
+            if (canGetLuckyBlock)
+            {
+                final int level = building.getBuildingLevel();
+                final LootParams lootParams = new Builder((ServerLevel) this.world)
+                                                .withParameter(LootContextParams.ORIGIN, position.getCenter())
+                                                .withParameter(LootContextParams.THIS_ENTITY, worker)
+                                                .withParameter(LootContextParams.TOOL, worker.getMainHandItem())
+                                                .withParameter(ModLootConditions.BUILDING_LEVEL, level)
```

Comment:
I don't like having this as a loot parameter; it makes things inconsistent with the way that recipe loot is handled.

For those, we deliberately have a separate loot table per level because it makes the JEI display nicer to be neatly separated.